File: 278-first-bad-version/278-first-bad-version.py
# The isBadVersion API is already defined for you.
# def isBadVersion(version: int) -> bool:

class Solution:
    def firstBadVersion(self, n: int) -> int:
        l = 1
        r = n
        res = 0
        while l <= r:
            mid = (l+r)//2
            if isBadVersion(mid):
                res = mid
                mid -= 1
                r = mid
            else:
                l = mid
                l += 1
        return res

File: 278-first-bad-version/278-first-bad-version.cpp
// The API isBadVersion is defined for you.
// bool isBadVersion(int version);

class Solution {
public:
    int firstBadVersion(int n) {
        long int l=0, r=n, res=0;
        while (l<=r){
            long int mid=(l+r)/2;
            if (isBadVersion(mid)){
                res=mid;
                r=mid-1;
            }
            else{
                l=mid+1;
            }
        }
        return res;
    }
};